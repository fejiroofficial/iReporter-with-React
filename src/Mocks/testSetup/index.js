/* eslint-disable import/no-extraneous-dependencies */
import '@babel/polyfill';
import React from 'react';
import { Provider } from 'react-redux';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import configMockStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';


export const mockState = {
  user: {
    token: '',
    isLoggedIn: false,
    loginStatus: false,
    message: '',
    isLoading: false,
    profile: {
      firstname: 'John',
      username: 'johnny',
      image: 'image.jpg',
      errorData: [],
    },
    resetPassword: {
      loading: false,
      message: '',
    }
  },
  bookmark: {
    loading: false,
    articles: [],
    message: '',
  },
  article: {
    newComments: {
      body: 'Hello world',
      id: 1,
      authorName: 'John',
      authorImage: 'image.jpg',
      date: 'July 4th 2018',
    },
    postingComment: false,
    commentMessage: 'Done',
    gettingComments: false,
    oldComments: [],
    hasMoreComments: false,
    commentPage: {
      current: 1,
      currentCount: 1,
      totalCount: 1,
    },
  },
  createArticle: {
    loading: false,
    message: false,
  },
  articlesReducer: {
    articles: []
  },
  articlesCategoryReducer: []
};


export const createMockStore = () => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  return mockStore;
};

const setup = (component, initailState = mockState) => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  const connectedWrapper = mount(
    <Provider store={mockStore(initailState)}>
      <BrowserRouter>
        {component}
      </BrowserRouter>
    </Provider>
  );
  return connectedWrapper;
};

export default setup;
