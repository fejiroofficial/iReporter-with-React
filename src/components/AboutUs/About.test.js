import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { createMockStore } from '../../Mocks/testSetup';
import AboutUs from './AboutUs';

createMockStore();

describe('<AboutUs>', () => {
  it('should render without crashing', () => {
    const container = (
      <AboutUs>
        <p>Hello</p>
      </AboutUs>
    );
    const wrapper = shallow(container);
    expect(wrapper).toMatchSnapshot();
  });
});