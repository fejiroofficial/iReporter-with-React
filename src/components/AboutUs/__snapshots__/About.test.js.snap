// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutUs> should render without crashing 1`] = `
<div>
  <div>
    WHO ARE WE
  </div>
  <div>
    ABOUT iREPORTER
  </div>
  <p>
    Corruption is a huge bane to Africaâ€™s development. African countries must develop novel and localised solutions that will curb this menace, hence the birth of iReporter. iReporter enables any/every citizen to bring any form of corruption to the notice of appropriate authorities and the general public. Users can also report on things that needs government intervention.
  </p>
  <h3>
    Communication
  </h3>
  <p>
    iReporter provides a platform for reporting corruption and intervention cases. Anyone can safely share what they experience and observe and can speak out against corruption. Our communication channels include our website, an SMS line, social media, email or post.
  </p>
  <h3>
    Investigation
  </h3>
  <p>
    We investigate selected reports of alleged acts of corruption, in particular those cases that have the most serious impact on our society. We hand our findings over to the relevant authorities to take further action, and we monitor the progress of each case. We work with mainstream and community media to make sure that corruption is fully exposed through our investigative work.
  </p>
</div>
`;
